src/Microsoft.AspNetCore.Hosting.Server.Abstractions/IServer.cs

```diff
@@ -22,5 +22,10 @@ public interface IServer : IDisposable
         /// <param name="application">An instance of <see cref="IHttpApplication{TContext}"/>.</param>
         /// <typeparam name="TContext">The context associated with the application.</typeparam>
         void Start<TContext>(IHttpApplication<TContext> application);
+
+        /// <summary>
+        /// Stop processing requests and shut down the server, gracefully if possible.
+        /// </summary>
+        void Stop();
```

Comment:
Consider passing in a timeout? And make it configurable like all the other WebHostOptions?